### PR TITLE
feat: OBAC changes

### PR DESCRIPTION
**Pull request by mohammadmuazam**

### CODE REVIEW

File: internal/api/squads.go [hunk 1]
```diff
@@ -46,36 +54,42 @@ func (client *Client) GetSquadByName(ctx context.Context, teamID string, name st
...
 	return RequestSlice[any, Squad](http.MethodGet, url, client, ctx, nil)
 }
 
 type CreateSquadReq struct {
-	Name      string   `json:"name"`
-	TeamID    string   `json:"owner_id"`
-	MemberIDs []string `json:"members"`
+	Name   string `json:"name"`
+	TeamID string `json:"owner_id"`
+	// MemberIDs []string `json:"members"`
```
Comment (Dineshsq) [inline internal/api/squads.go]: remove this line?

File: internal/provider/resource_team_member.go [hunk 2]
```diff
@@ -81,10 +86,26 @@ func resourceTeamMemberImport(ctx context.Context, d *schema.ResourceData, meta
...
+
+	if len(roleIDs) > 0 && len(role) > 0 {
+		return diag.Errorf("role_ids and role cannot be passed")
+	}
+
+	if len(roleIDs) > 0 {
+		createReq.RoleIDs = roleIDs
+	}
+
+	if len(role) > 0 {
```
Comment (Dineshsq) [inline internal/provider/resource_team_member.go]: if strings.TrimSpace(role)!=""

File: internal/provider/resource_team_member.go [hunk 3]
```diff
@@ -115,10 +136,24 @@ func resourceTeamMemberRead(ctx context.Context, d *schema.ResourceData, meta an
...
+
+	if len(roleIDs) > 0 && len(role) > 0 {
+		return diag.Errorf("role_ids and role cannot be passed")
+	}
+
+	if len(roleIDs) > 0 {
+		updateReq.RoleIDs = roleIDs
+	}
+
+	if len(role) > 0 {
```
Comment (Dineshsq) [inline internal/provider/resource_team_member.go]: if strings.TrimSpace(role)!=""

**Review by Dineshsq — CHANGES_REQUESTED** [review]

File: docs/resources/squad.md [hunk 4]
```diff
@@ -20,10 +20,40 @@ data "squadcast_team" "example_team" {
...
+  members {
+      user_id = data.squadcast_user.example_user.id
+  }
+  members {
+      user_id = data.squadcast_user.example_user_2.id
+  }
+}
+
+# OBAC permission model
+resource "squadcast_squad" "example_squad_rbac" {
```
Comment (roshan8) [inline docs/resources/squad.md]: example_squad_rbac => example_squad_obac

File: internal/provider/resource_squad.go [hunk 5]
```diff
@@ -132,10 +180,43 @@ func resourceSquadRead(ctx context.Context, d *schema.ResourceData, meta any) di
...
-	})
+	updateReq := &api.UpdateSquadReq{
+		Name: d.Get("name").(string),
+	}
+	memberIDs := tf.ListToSlice[string](d.Get("member_ids"))
+	members := d.Get("members").([]interface{})
+
+	// if len(members) > 0 && len(memberIDs) > 0 {
+	// 	return diag.Errorf("member_ids and members cannot be passed at once")
+	// }
```
Comment (roshan8) [inline internal/provider/resource_squad.go]: remove this

**Review by roshan8 — CHANGES_REQUESTED** [review]

**Review by Dineshsq — APPROVED** [review]
This might have breaking changes, please follow semantic versioning.

**Review by roshan8 — APPROVED** [review]

**Pull request merged**